util/gitutil/gitutil.go

```diff
@@ -134,3 +134,7 @@ func (c *Git) clean(out string, err error) (string, error) {
 	}
 	return out, err
 }
+
+func IsUnknownRevision(err error) bool {
+	return err != nil && (strings.Contains(err.Error(), "unknown revision or path not in the working tree") || strings.Contains(err.Error(), "bad revision"))
```

Comment:
Is this error produced by git itself? Would it make sense to link to the source code where this originates from (in case this error would ever change; that way we know where to look for the updated error string).

(I did a quick search, and found other projects also referring to this error, so I "assume" it's an error produced by git https://grep.app/search?q=%22unknown%20revision%20or%20path%20not%20in%20the%20working%20tree%22)

Reply:
LOL, looks like others [generate this errors with a capital `U`](https://github.com/github/hub/blob/38bcd4ae469e5f53f01901340b715c7658ab417a/git/git.go#L136) (do we need to to `strtolower` the error, or match on `nknown revision` (skipping the first letter)? 

I also wonder if this string can be localised on Windows; looking at this line, (using `test_i18ngrep `) I fear that may be possible; https://github.com/git/git/blob/a6a323b31e2bcbac2518bddec71ea7ad558870eb/t/t1506-rev-parse-diagnosis.sh#L112

Reply:
Maybe we need to call `git` with a specific locale 🤔 https://stackoverflow.com/a/13044660/1811501

Reply:
Might be better to use `LC_ALL=C` for git commands then https://www.gnu.org/software/gettext/manual/html_node/The-LANGUAGE-variable.html to override all the other localization settings.

Reply:
LOL, yeah, whatever works (and whatever doesn't _require_ locale configs to be present) as long as we get something consistent that we can parse.

My eye dropped on it, and I recall users on Windows reporting tickets and in those tickets various commands were translated, so I anticipated "that could be problematic if we're looking for a specific string"